First Q

```diff
@@ -0,0 +1,95 @@
+# COMP3410 Program Template
+# Author: Your Name Dereje Arega 
+# Assignment: PA1
+# Date: 2/11/2015
+
+# Turn in one .asm file per assignment component
+# Remember to submit it as a pull request to the GitHub repo for the assignment
+
+# fibonacci.asm
+# Input: None
+# Output: Print to terminal N Fibonacci numbers
+
+##########################################################
+# .data segment. Constant and variable definitions go here.
+##########################################################
+
+	.data
+
+
+fibs:	.word   0 : 9         # create an array variable named "fibs" of 9 word-length elements (4 bytes each)
+size:	.word  9              # create a single integer variable named "size" that indicates the length of the array
+text: .asciiz "How many Fibonacci numbers you want print?\n"
+###############################################
+# .text segment. Assembly instructions go here.
+###############################################
+	.text
+	      la   $s0, fibs        # load address of array into $s0
+	      #la   $s5, size        # load address of size variable into $s5
+	     # lw   $s5, 0($s5)      # load array size from its address in the register
+	     li $v0,4
+	     la $a0, text
+	     syscall
+		li $v0,5
+		syscall
+		move $s5,$v0
```

Comment:
-5 Doesn't limit the user to 9 and therefore breaks on relatively small input.  This is unsafe because your loop is accessing addresses that are out of the bounds set in fibs.  (For my test, output is invalid for input > 21 where if you allocate more memory you will see that it works fine)

In fact if you try to run it with 25 numbers and pay attention to $a0 after it loads the "space" memory address, you may see the value 268501076, which is where the space character is stored, but for 25 fibonacci numbers, the sequence ends at 268501084, which means that the space character " " has been replaced with another character.